packages/flutter/lib/src/material/about.dart

```diff
@@ -556,49 +844,123 @@ class _LicensePageState extends State<LicensePage> {
   @override
   Widget build(BuildContext context) {
     assert(debugCheckHasMaterialLocalizations(context));
-    final String name = widget.applicationName ?? _defaultApplicationName(context);
-    final String version = widget.applicationVersion ?? _defaultApplicationVersion(context);
-    final Widget icon = widget.applicationIcon ?? _defaultApplicationIcon(context);
-    final MaterialLocalizations localizations = MaterialLocalizations.of(context);
-    return Scaffold(
-      appBar: AppBar(
-        title: Text(localizations.licensesPageTitle),
-      ),
-      // All of the licenses page text is English. We don't want localized text
-      // or text direction.
-      body: Localizations.override(
-        locale: const Locale('en', 'US'),
-        context: context,
-        child: DefaultTextStyle(
-          style: Theme.of(context).textTheme.caption,
-          child: SafeArea(
-            bottom: false,
-            child: Scrollbar(
-              child: ListView(
-                padding: const EdgeInsets.symmetric(horizontal: 8.0, vertical: 12.0),
-                children: <Widget>[
-                  Text(name, style: Theme.of(context).textTheme.headline5, textAlign: TextAlign.center),
-                  if (icon != null) IconTheme(data: Theme.of(context).iconTheme, child: icon),
-                  Text(version, style: Theme.of(context).textTheme.bodyText2, textAlign: TextAlign.center),
-                  Container(height: 18.0),
-                  Text(widget.applicationLegalese ?? '', style: Theme.of(context).textTheme.caption, textAlign: TextAlign.center),
-                  Container(height: 18.0),
-                  Text('Powered by Flutter', style: Theme.of(context).textTheme.bodyText2, textAlign: TextAlign.center),
-                  Container(height: 24.0),
-                  ..._licenses,
-                  if (!_loaded)
-                    const Padding(
-                      padding: EdgeInsets.symmetric(vertical: 24.0),
-                      child: Center(
-                        child: CircularProgressIndicator(),
+    final String package = widget.packageName;
+    final MaterialLocalizations localisations =
+        MaterialLocalizations.of(context);
+    final double gutterSize = _getGutterSize(context);
+
+    if (widget.scrollController == null) {
```

Comment:
there's a lot of duplicated content here, might be worth seeing if there's a way to factor it out so the branches are only the pieces that differ